tasks/devcontainer.py

```diff
@@ -166,3 +169,39 @@ def is_up(ctx) -> bool:
 def is_installed(ctx) -> bool:
     res = ctx.run("which devcontainer", hide=True, warn=True)
     return res.ok
+
+
+def run_on_devcontainer(func):
+    """
+    This decorator will run the decorated function in a devcontainer if the selected platform is linux.
+    All you need to do is to decorate your task with this decorator and add a `platform` argument to your task.
+    """
+
+    @wraps(func)
+    def _run_on_devcontainer(ctx, *args, **kwargs):
+        if kwargs.get('platform'):
+            platform = kwargs['platform'].lower()
+            if platform == "linux" and py_platform.system().lower() != platform:
+                # If we choose to run them on linux, and we are not on linux already
+                if not file().exists():
+                    print(color_message("Generating the devcontainer file to run the linter in a container.", "orange"))
+                    # TODO remove the hardcoded image and auto-pull it
+                    setup(ctx, image="486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-devenv:1-arm64")
+
+                if not is_up(ctx):
+                    print(color_message("Starting the devcontainer...", "orange"))
+                    start(ctx)
+
+                print(color_message("Running the command in the devcontainer...", "orange"))
+                cli = DockerCLI(DEVCONTAINER_NAME)
+
+                cmd = ["inv"] + sys.argv[1:]
```

Comment:
Can we have a dead loop here? Looks like we will call
`inv linter.go --platform linux` in the devcontainer, what happens in this case?